### DISCUSSION TITLE
How to setup storage for onedirve ?

### DISCUSSION THREAD

**Discussion by jigujigu** [filamentphp/filament#6444] [Help]
Hi All,
I am trying to upload to onedrive storage. 
```
                Forms\Components\FileUpload::make('attachment')
                ->label('attachment')
                ->disk('onedrive')
                ->preserveFilenames()
```

I already create AccessToken  form certificate & secret in active directory menu

![image](https://user-images.githubusercontent.com/6214964/236672064-734ef8b1-82a5-4b72-b7b0-fe95dfbf11fe.png)

and put the result in .env file
```
ONEDRIVE_ROOT="D6497FA6A189815D"
ONEDRIVE_ACCESS_TOKEN=ca8dd377-2cbe-4ceb-aa23-fffe3e80facd
ONEDRIVE_DIR_TYPE=drives
```

and added these line into config\filesystem.php
```
        'onedrive' => [
            'driver' => 'onedrive',
            'root' => env('ONEDRIVE_ROOT'),
            'access_token' => env('ONEDRIVE_ACCESS_TOKEN'), //optional if demanded
            'directory_type' => env('ONEDRIVE_DIR_TYPE')
        ],
```

and added this line in config\app
`    Justus\FlysystemOneDrive\Providers\OneDriveAdapterServiceProvider::class,`

and install the package
`composer require justus/flysystem-onedrive`

But the upload is failed with error code (on browser) :
http://127.0.0.1:8000/livewire/upload-file?expires=1683424928&signature=ae31676c34d78a2be2fa4b73458d353014f72ecfccd83468f0cc9217b2ed1a52 500 (Internal Server Error)

Is anyone have experience using one drive as laravel storage ?

thank you.

**Comment by danharrin** [comment]
Livewire only supports S3-compatible filesystems for uploads.
